pkgs/development/tools/zigmod/default.nix

```diff
@@ -0,0 +1,163 @@
+    }
+    {
+      url = "https://github.com/nektro/zig-inquirer";
+      rev = "14c3492c46f9765c3e77436741794d1a3118cbee";
+      sha256 = "sha256-62+paPZiuyFdrV/TOZjec2y/sRgPtAvE36SLFsaSevo=";
+    }
+    {
+      url = "https://github.com/nektro/zig-json";
+      rev = "a091eaa9f9ae91c3875630ba1983b33ea04971a3";
+      sha256 = "sha256-iFrmpzD92wL+ldv/6ZKCtcNuNFkhcw56i5Yolp4xE8U=";
+    }
+    {
+      url = "https://github.com/nektro/zig-leven";
+      rev = "ab852cf74fa0b4edc530d925f0654b62c60365bf";
+      sha256 = "sha256-XUPag1aHtC+yMm0loqgHESyN8ElUWGI7GT5DLN8vZe4=";
+    }
+    {
+      url = "https://github.com/nektro/zig-licenses";
+      rev = "c9b8cbf3565675a056ad4e9b57cb4f84020e7680";
+      sha256 = "sha256-lnqDt9yaK2Ed3W14vmxcd5kraG/QDSh2rSmvC/5naiU=";
+    }
+    {
+      url = "https://github.com/nektro/zig-licenses-text";
+      rev = "3c07c6e4eb0965dafd0b029c632f823631b3169c";
+      sha256 = "sha256-Jygr1u2HYSRZNfq/aH+chaeV608dmvLIIPiFifKOWzY=";
+    }
+    {
+      url = "https://github.com/nektro/zig-range";
+      rev = "4b2f12808aa09be4b27a163efc424dd4e0415992";
+      sha256 = "sha256-Jmz2GFieXB4CALZ6XqMq9LbhkNoxF53/ovodQlg6PAM=";
+    }
+    {
+      url = "https://github.com/nektro/zig-time";
+      rev = "52a6050939b83e9f8b52b2236a36332dcb4c17a3";
+      sha256 = "sha256-g7Gmq7ZZXBB26rFezuN+lzmdm03WyPakblrkfanLPhM=";
+    }
+    {
+      url = "https://github.com/truemedian/hzzp";
+      rev = "d4fbade2d806fc93bc5f79ec8efdcc25ea625fa3";
+      sha256 = "sha256-7F92vl+5r5er36NaX/THxQ3qLRRUdAs52PW+ZeiSrZI=";
+    }
+    {
+      url = "https://github.com/truemedian/zfetch";
+      rev = "829973144f680cd16be16923041fa810c1dee417";
+      sha256 = "sha256-H3kEZ+M/OTgkC6gUjN5KHzCkd2bE5mLlES/RTT0NitI=";
+    }
+    {
+      url = "https://github.com/ziglibs/known-folders";
+      rev = "24845b0103e611c108d6bc334231c464e699742c";
+      sha256 = "sha256-xMgdOKwWqBmw7avcioqQQrrPU1MjzlBMtNjqPfOEtDQ=";
+    }
+    {
+      url = "https://github.com/yaml/libyaml";
+      rev = "2c891fc7a770e8ba2fec34fc6b545c672beb37e6";
+      sha256 = "sha256-S7PnooyfyAsIiRAlEPGYkgkVACGaBaCItuqOwrq2+qM=";
+    }
+  ];
+
+  copyDep = { url, rev, sha256 }:
+    let
+      dir_path = lib.removePrefix "https://" url;
+      store_path = fetchgit { inherit url rev sha256; };
+    in
+    ''
+      dep_path=$sourceRoot/zig-cache/zigmod/deps/git/${dir_path}/${rev}
+      mkdir -p $dep_path
+      cp -r ${store_path}/* $dep_path
+    '';
+
+  copyDeps = lib.concatMapStrings copyDep deps;
+in
+stdenv.mkDerivation rec {
+  pname = "zigmod";
+  version = "r83";
+
+  src = fetchFromGitHub {
+    owner = "nektro";
+    repo = "zigmod";
+    rev = version;
+    sha256 = "1sprzjylyjifqvakz8cm0ghmiqzvrsvkyhw49yy6qb3i2pi6jwiz";
+  };
+
+  nativeBuildInputs = [
+    zig
+  ];
+
+  preUnpack = ''
+    rm -rf source
+  '';
+
+  postUnpack = ''
+    ${copyDeps}
+  '';
+
+  preBuild = ''
+    export XDG_CACHE_HOME="$(mktemp -d)"
```

Comment:
We should put this into a setup hook for zig in another PR.

Reply:
Is that a prerequisite? Otherwise, I can add a `FIXME` comment here if you'd like.

Reply:
He is just commenting about the Zig compiler, not your package.